universe.cpp

```diff
@@ -386,3 +387,165 @@ void Universe::updateTriangleData() {
 		triangleNeighbors.at(t) = {t->getTriangleLeft(), t->getTriangleRight(), t->getTriangleCenter()};
 	}
 }
+
+void Universe::exportGeometry(std::string geometryFilename) {
+
+	updateTriangleData();
+	updateVertexData();
+
+	std::unordered_map<int, int> vertexMap;
+	std::vector<Vertex::Label> intVMap;
+	intVMap.resize(vertices.size());
+
+	int i = 0;
+	for (auto v : vertices) {
+		vertexMap.insert({v, i});
+		intVMap.at(i) = v;
+		i++;
+	}
+
+	std::unordered_map<int, int> triangleMap;
+	std::vector<Triangle::Label> intTMap;
+	intTMap.resize(triangles.size());
+
+	i = 0;
+	for (auto t : triangles) {
+		triangleMap.insert({t, i});
+		intTMap.at(i) = t;
+		i++;
+	}
+
+	std::string output;
+
+	output += std::to_string(vertices.size());
+	output += "\n";
+
+	for (int j = 0; j < intVMap.size(); j++) {
+		output += std::to_string(intVMap.at(j)->time);
+		output += "\n";
+	}
+
+	output += std::to_string(vertices.size());
+	output += "\n";
+	output += std::to_string(triangles.size());
+	output += "\n";
+
+	for (int j = 0; j < intTMap.size(); j++) {
+		Triangle::Label t = intTMap.at(j);
+
+		Vertex::Label tVs [3] = {t->getVertexLeft(),t->getVertexRight(),t->getVertexCenter()};
+		for (auto v : tVs) {
+			output += std::to_string(vertexMap.at(v));
+			output += "\n";
+		}
+
+		Triangle::Label tNeighb [3] = {t->getTriangleLeft(),t->getTriangleRight(),t->getTriangleCenter()};
+		for (auto t : tNeighb) {
+			output += std::to_string(triangleMap.at(t));
+			output += "\n";
+		}
+	}
+
+	output += std::to_string(triangles.size());
+
+    std::ofstream file;
+    file.open(geometryFilename, std::ios::out | std::ios::trunc);
+	assert(file.is_open());
+
+	file << output << "\n";
+	file.close();
+
+    std::cout << geometryFilename << "\n";
+}
+
+void Universe::importGeometry(std::string geometryFilename) {
+	trianglesAll.clear(); //added function to bag!
+	verticesFour.clear();
+	trianglesFlip.clear();
+	sliceSizes.clear();
```

Comment:
these four lines can be deleted — at the moment we don't want to allow for an import to occur after a Universe has already be constructed, so the expectation at this stage should be that all Bags and the vector sliceSizes are empty.